pkg/pagerduty/reciever/reciever.go

```diff
@@ -0,0 +1,55 @@
+package reciever
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+
+	sdk "github.com/PagerDuty/go-pagerduty"
+)
+
+// fileReader will wrap the os or fstest.MapFS stucts so we are not locked in
+type fileReader interface {
+	ReadFile(name string) ([]byte, error)
+}
+
+// ExtractExternalIDFromPayload will retrieve the payloadPath and return the externalID
+func ExtractExternalIDFromPayload(payloadFilePath string, reader fileReader) (string, error) {
+	data, err := readPayloadFile(payloadFilePath, reader)
+	if err != nil {
+		return "", fmt.Errorf("could not read the payloadFile: %w", err)
+	}
+
+	p := sdk.WebhookPayload{}
+	err = json.Unmarshal(data, p)
+	if err != nil {
+		return "", UnmarshalErr{Err: err}
+	}
+	//incidentID := p.Incident.IncidentKey
+
+	return "", nil
+}
+
+func readPayloadFile(payloadFilePath string, reader fileReader) ([]byte, error) {
+	fileNotFoundErr := FileNotFoundErr{FilePath: payloadFilePath}
```

Comment:
Did you check whether that in this case (meaning payloadFilePath==""),  doesn't already fall into the PathError of reader.ReadFile?

Reply:
maybe?
I'll try and remove the check and see if this works still

Reply:
Yes, I would check that because if this is the case you are basically defining work twice since it should be also done by the reader library.